workflows/prognostic_scream_run/scream_run/steppers/machine_learning.py

```diff
@@ -0,0 +1,104 @@
+import dataclasses
+from typing import (
+    Hashable,
+    Iterable,
+    MutableMapping,
+    Mapping,
+    Optional,
+    Sequence,
+    Set,
+    cast,
+)
+import fv3fit
+import xarray as xr
+import vcm
+import os
+
+
+State = MutableMapping[Hashable, xr.DataArray]
+
+
+@dataclasses.dataclass
+class MachineLearningConfig:
+    """Machine learning configurations
+
+    Attributes:
+        model: list of URLs to fv3fit models.
+        diagnostic_ml: do not apply ML tendencies if true.
+        scaling: if given, scale the outputs by the given factor. This is a manually
+            defined alteration of the model, and should not be used for
+            normalization.
+    Example::
+
+        MachineLearningConfig(
+            model=["gs://vcm-ml-scratch/test-annak/ml-pipeline-output"],
+            diagnostic_ml=False,
+        )
+
+    """
+
+    model: Sequence[str] = dataclasses.field(default_factory=list)
+    diagnostic_ml: bool = False
+    scaling: Mapping[str, float] = dataclasses.field(default_factory=dict)
+
+
+class MultiModelAdapter:
+    def __init__(
+        self,
+        models: Iterable[fv3fit.Predictor],
+        scaling: Optional[Mapping[str, float]] = None,
+    ):
+        """
+        Args:
+            models: models for which to combine predictions
+            scaling: if given, scale the predictions by the given factor
+        """
+        self.models = models
+        if scaling is None:
+            self._scaling: Mapping[str, float] = {}
+        else:
+            self._scaling = scaling
+
+    @property
+    def input_variables(self) -> Set[str]:
+        vars = [model.input_variables for model in self.models]
+        return {var for model_vars in vars for var in model_vars}  # type: ignore
```

Comment:
```suggestion
        return {
            var 
            for model in self.models 
            for var in model.input_variables
        }  # type: ignore
```

Reply:
Just trying to improve the clarity here since the return is hard to parse.   Another idea:

```
        all_inputs = []
        for model in self.models:
            all_inputs.extend(model.input_vars)
        return set(all_inputs)
```

Reply:
I like this too. Definitely more clear this way.